examples/step-67/doc/intro.dox

```diff
@@ -0,0 +1,680 @@
+the Laplacian of step-37, it now returns a type that is
+`Tensor<1,dim+2,VectorizedArray<Number>>`. Likewise, the gradient type is now
+`Tensor<1,dim+2,Tensor<1,dim,VectorizedArray<Number>>>`, where the outer
+tensor collects the `dim+2` components of the Euler system, and the the inner
+tensor the partial derivatives in the various directions. For example, the
+flux of the Euler system is of this type. In order to reduce the amount of
+code we have to write for spelling out these types, we use the C++ `auto`
+keyword where possible.
+
+From an implementation point of view, the nonlinearity is not a big
+difficulty: It is introduced naturally as we express the terms of the Euler
+weak form, for example in the form of the momentum term $\rho \mathbf{u}
+\otimes \mathbf{u}$. To obtain this expression, we first deduce the velocity
+$\mathbf{u}$ from the momentum variable $\rho \mathbf{u}$. Given that $\rho
+\mathbf{u}$ is represented as a $p$-degree polynomial, as is $\rho$, the
+velocity $\mathbf{u}$ is a rational expression in terms of the reference
+coordinates $\hat{\mathbf{x}}$. As we perform the multiplication $(\rho
+\mathbf{u})\otimes \mathbf{u}$, we obtain an expression that contains a
+rational expression of two polynomials, with polynomial degree $2p$ in the
+numerator and polynomial degree $p$ in the denominator. Combined with the
+gradient of the test function, the integrand is of degree $3p$ in the
+numerator and $p$ in the denominator already for affine (undeformed) cell
+geometries. For curved cells, additional polynomial and rational expression
+appear. At this point, one usually needs to give up on insisting on exact
+integration, and take whatever accuracy the Gaussian (more precisely,
+Gauss--Legrende) quadrature provides. As that formulas only integrate
+polynomials exactly, we have to live with the variational crime in the form of
+an integration error.
+
+While inaccurate integration is usually tolerable for elliptic problems, for
+hyperbolic problems inexact integration causes some headache in the form of an
+effect called <b>aliasing</b>. The term comes from signal processing and
+expresses the situation of inappropriate, too coarse sampling. In terms of
+quadrature, the inappropriate sampling means that we use too few quadrature
+points as what would be required to accurately sample the variable-coefficient
+integrand. It has been shown in DG literature that aliasing errors can
+introduce unphysical oscillations in the numerical solution for <i>barely</i>
+resolved simulations. The fact that aliasing mostly effects coarse resolutions
+-- whereas finer meshes with the same scheme otherwise work fine -- is not
+surprising because well-resolved simulations have small coefficients in the
+higher polynomial degrees that are missed by too few quadrature points,
+whereas the main solution contribution in the lower polynomial degrees is
+still well-captured. To address this topic, various approaches have been
```

Comment:
In other words, Taylor's theorem helps you! On short enough length scales, every function looks linear and can easily be integrated accurately :-)